src/transaction/log_replication_atomic.cpp

```diff
@@ -507,21 +530,34 @@ namespace cublog
   void
   atomic_replicator::discard_caches_for_ddl (cubthread::entry &thread_entry, const TRANID trid)
   {
-    assert (m_locked_classes.count (trid) > 0);
-
-    for (auto classoid : m_locked_classes[trid])
+    auto [begin, end] = m_locked_classes.equal_range (trid);
+    for (auto it = begin; it != end; ++it)
       {
-	update_classname_cache_for_ddl (thread_entry, &classoid);
-	(void) heap_classrepr_decache (&thread_entry, &classoid);
-	(void) heap_delete_hfid_from_cache (&thread_entry, &classoid);
-	xcache_remove_by_oid (&thread_entry, &classoid);
-	partition_decache_class (&thread_entry, &classoid);
+	auto log_rec = it->second;
+
+	update_classname_cache_for_ddl (thread_entry, &log_rec.oid);
+	(void) heap_classrepr_decache (&thread_entry, &log_rec.oid);
+	(void) heap_delete_hfid_from_cache (&thread_entry, &log_rec.oid);
+	xcache_remove_by_oid (&thread_entry, &log_rec.oid);
+	partition_decache_class (&thread_entry, &log_rec.oid);
       }
   }
 
   bool
-  atomic_replicator::is_locked_for_ddl (const TRANID trid, const OID *classoid)
+  atomic_replicator::is_locked_for_ddl (const TRANID trid, const OID *oid, const bool is_class)
   {
-    return m_locked_classes.count (trid) > 0 && m_locked_classes[trid].count (*classoid) > 0;
+    const auto locked_objects = is_class ? m_locked_classes : m_locked_serials;
```

Comment:
```suggestion
    const auto& locked_objects = is_class ? m_locked_classes : m_locked_serials;
```
Otherwise, isn't it a copy of the container?

Also, `const` function.

Reply:
Oh.. thanks